Diffusion regression tests based on hydra config

GMNGeoffrey opened the pull request:
The existing tests that run the examples have the advantage that they can reuse something that already exists and they ensure the user-facing examples don't break. But for just expanding test coverage, it's a lot nicer and less error prone to write specs directly IMO. I want to use a similar setup for benchmarking as well.

tpopp commented on tests/test_diffusion.py:
```diff
@@ -125,6 +135,22 @@ def test_command(script, tmp_path, reference_dir, child_env, request):
+    output_dir = tmp_path
+    # The pytest docs say you need to create this directory, but empirically it
+    # is already created.
+    # output_dir.mkdir()
```
Same nit on removing the commented code as my personal preference.

GMNGeoffrey committed:
Diffusion regression tests based on hydra configs

The existing tests that run the examples have the advantage that they
can reuse something that already exists and they ensure the user-facing
examples don't break. But for just expanding test coverage, it's a lot
nicer and less error prone to write specs directly IMO. I want to use a
similar setup for benchmarking as well.

I think there might be a more idiomatic hydra way to do this, but
looking through their docs I didn't find an obviously good one.

For now, this is just a couple of tests of unconditional generation, but
it's set up to be expanded to more interesting cases.
